// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package safeentry;

// The greeting service definition.
service SafeEntryService {
  rpc Checkin (CheckIn_Request) returns (CheckIn_Reply) {}
  rpc Checkout (Request) returns (Reply) {}
  rpc History (History_Request) returns (History_Reply) {}
  rpc Contacted (Request) returns (Reply) {}
  rpc checkContacted (Check_Request) returns (Check_Reply) {}
}

// The request message containing check in details
message CheckIn_Request {
  string name = 1;
  string nric = 2;
  string location = 3;
  string datetime = 4;
  int32 groupid = 5;

}

// The response message containing status of check in and details.
message CheckIn_Reply {
  string message = 1;
}

message Request {
  string name = 1;
  string nric = 2;
  string location = 3;
  string datetime = 4;
}

message Reply {
  string message = 1;
}

message History_Request {
  string nric = 1;
}

message History_Reply {
  repeated History_Item histories = 1;
}

message History_Item {
  string name = 1;
  string nric = 2;
  string location = 3;
  string checkin_dt = 4;
  string checkout_dt = 5;
}

// The request message containing status of check in and details.
message Check_Request {
  string nric = 1;
}

// The response message containing status of check in and details.
message Check_Reply {
  repeated Check_item checks = 1;
}

// The check item
message Check_item {
  string name = 1;
  string nric = 2;
  string location = 3;
  string checkin_dt = 4;
  string checkout_dt = 5;
}